lib/portlayer/exec/container.go

```diff
@@ -151,6 +151,20 @@ func (s State) String() string {
 	return ""
 }
 
+// CurrentState returns current state.
+func (c *Container) CurrentState() State {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.state
+}
+
+// SetState changes container state.
+func (c *Container) SetState(s State) {
```

Comment:
Come to think of it, this should not be an exported method on the Container object. The only way you can change the state of the container is through the handle.SetState, followed by Commit.